substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1,94 +1,164 @@
-    ensure, log,
-    traits::{
-        Currency, ExistenceRequirement::KeepAlive, Get, LockIdentifier, LockableCurrency,
-        WithdrawReasons,
-    },
+    dispatch::DispatchErrorWithPostInfo,
+    ensure,
+    traits::{Currency, ExistenceRequirement, Get, LockableCurrency, WithdrawReasons},
     weights::Pays,
 };
 use frame_system::{self as system, ensure_signed};
-use pallet_balances;
 use pallet_tfgrid;
 use pallet_tfgrid::types as pallet_tfgrid_types;
 use pallet_tft_price;
 use pallet_timestamp as timestamp;
 use sp_runtime::{
     traits::{CheckedSub, SaturatedConversion},
-    DispatchError, DispatchResult, Perbill, Percent,
+    Perbill, Percent,
 };
-use sp_std::convert::TryInto;
-use sp_std::vec::Vec;
 use substrate_fixed::types::U64F64;
 use tfchain_support::{
     traits::ChangeNode,
-    types::{Node, NodeCertification, PublicIP, Resources},
+    types::{Node, NodeCertification, Resources},
 };
 
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
 #[cfg(test)]
 mod mock;
 
 #[cfg(test)]
 mod tests;
 
-mod benchmarking;
-
-pub mod types;
 pub mod weights;
 
-pub use weights::WeightInfo;
-
-pub const GRID_LOCK_ID: LockIdentifier = *b"gridlock";
-
-pub trait Config:
-    system::Config + pallet_tfgrid::Config + pallet_timestamp::Config + pallet_balances::Config
-{
-    type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
-    type Currency: LockableCurrency<Self::AccountId>;
-    type StakingPoolAccount: Get<Self::AccountId>;
-    type BillingFrequency: Get<u64>;
-    type DistributionFrequency: Get<u16>;
-    type GracePeriod: Get<u64>;
-    type WeightInfo: WeightInfo;
-    type NodeChanged: ChangeNode;
-}
-
-pub const CONTRACT_VERSION: u32 = 3;
-
-pub type BalanceOf<T> =
-    <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+pub mod types;
 
-decl_event!(
-    pub enum Event<T>
-    where
-        AccountId = <T as frame_system::Config>::AccountId,
-        BalanceOf = BalanceOf<T>,
+// Definition of the pallet logic, to be aggregated at runtime definition
+// through `construct_runtime`.
+#[frame_support::pallet]
+pub mod pallet {
+    // use hex::FromHex;
+    use super::types::*;
+    use super::weights::WeightInfo;
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_support::{
+        dispatch::DispatchResultWithPostInfo,
+        log,
+        traits::{Currency, Get, LockIdentifier, LockableCurrency},
+    };
+    use frame_system::pallet_prelude::*;
+    use sp_std::convert::TryInto;
+    use sp_std::vec::Vec;
+    use tfchain_support::{traits::ChangeNode, types::PublicIP};
+
+    pub type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+
+    pub const GRID_LOCK_ID: LockIdentifier = *b"gridlock";
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
```

Comment:
Why without storage info?

Reply:
Storage information requires you to have bounds on Vectors and implement MaxEncodedLen. I will add this in a later version since this requires some extra work. Will open an issue for it